core/src/com/unciv/ui/civilopedia/CivilopediaText.kt

```diff
@@ -8,28 +8,66 @@ import com.badlogic.gdx.utils.Align
-/** Represents a text line with optional linking capability.
+/** Represents a decorated text line with optional linking capability.
+ *  A line can have up to three icons: link, object, star in that order.
  *  Special cases:
+ *  - Standalone image from atlas or from ExtraImages
+ *  - A separator line ([separator])
  *  - Automatic external links (no [text] but [link] begins with a URL protocol)
  * 
  * @param text          Text to display.
  * @param link          Create link: Line gets a 'Link' icon and is linked to either 
  *                      an Unciv object (format `category/entryname`) or an external URL.
+ * @param icon          Display an Unciv object's icon inline but do not link (format `category/entryname`).
  * @param extraImage    Display an Image instead of text. Can be a path as understood by
  *                      [ImageGetter.getImage] or the name of a png or jpg in ExtraImages.
  * @param imageSize     Width of the [extraImage], height is calculated preserving aspect ratio. Defaults to available width.
+ * @param size          Text size, defaults to 18f. Use [size] or [header] but not both.
  * @param header        Header level. 1 means double text size and decreases from there.
- * @param separator     Renders a separator line instead of text.
+ * @param indent        Indentation - 0 = text will follow icons with a little padding,
+ *                      1 = aligned to a little more than 3 icons, each step above that adds 30f.
+ * @param padding       Defines vertical padding between rows, defaults to 5f.
+ * @param color         Sets text color, accepts Java names or 6/3-digit web colors (e.g. #FFA040).
+ * @param separator     Renders a separator line instead of text. Can be combined only with color and size (the latter being line width, defaulting to 2)
+ * @param italic        Renders text in italic font style (not implemented)
+ * @param bold          Renders text in bold font style (not implemented)
+ * @param strike        Renders text in strikethrough font style (not implemented)
+ * @param shadow        Renders text with a dark shadow (not implemented)
+ * @param starred       Decorates text with a star icon - if set, it receives the [color] instead of the text.
+ * @param centered      Centers the line (and turns off wrap)
  */
 class FormattedLine (
     val text: String = "",
     val link: String = "",
+    val icon: String = "",
     val extraImage: String = "",
     val imageSize: Float = Float.NaN,
+    val size: Int = Int.MIN_VALUE,
     val header: Int = 0,
+    val indent: Int = 0,
+    val padding: Float = Float.NaN,
+    val color: String = "",
     val separator: Boolean = false,
+    val italic: Boolean = false,        // Not implemented - would need separate font
+    val bold: Boolean = false,          // Not implemented - see 'MSDF'?
+    val strike: Boolean = false,        // Not implemented - could do a simple line
+    val shadow: Boolean = false,        // Not implemented - see 'MSDF'.
```

Comment:
Anything that's not implemented should not be added to the class yet. This is just more unnecessary code for 'stuff that should be' and 'whoops, you used something that isn't implemented'. This doesn't help at all.

Reply:
:+1: Understood. In that case _preparing_ (but not implementing) mod checker to tell us about incorrect uses goes shelved, too, for now.